deeplake/core/vectorstore/deeplake_vectorstore.py

```diff
@@ -0,0 +1,213 @@
+import deeplake
+from deeplake.core.vectorstore import utils, vector_search
+
+import logging
+import uuid
+from functools import partial
+from typing import Optional, Any, Iterable, List, Dict, Callable
+
+import numpy as np
+
+
+logger = logging.getLogger(__name__)
+
+
+class DeepLakeVectorStore:
+    """Base class for DeepLakeVectorStore"""
+
+    _DEFAULT_DEEPLAKE_PATH = "./deeplake_vector_store"
+
+    def __init__(
+        self,
+        dataset_path: str = _DEFAULT_DEEPLAKE_PATH,
+        token: Optional[str] = None,
+        embedding_function: Optional[callable] = None,
+        read_only: Optional[bool] = False,
+        ingestion_batch_size: int = 1024,
+        num_workers: int = 4,
```

Comment:
default let's set to 0 because ingest transform fails if embedding is not an api but a e.g. HuggingFace model.